src/BatchRequestContent.ts

```diff
@@ -0,0 +1,312 @@
+     */
+    static async getJSONFromRequest(request: Request): Promise<any> {
+        let requestJSON: KeyValuePairObject = {};
+        // Stripping off hostname, port and url scheme
+        requestJSON.url = "/" + request.url.split(/.*?\/\/.*?\//)[1];
+        requestJSON.method = request.method;
+        let headers = {};
+        for (const pair of request.headers.entries()) {
+            headers[pair[0]] = pair[1];
+        }
+        requestJSON.headers = headers;
+        if (request.method === RequestMethod.PATCH || request.method === RequestMethod.POST || request.method === RequestMethod.PUT) {
+            requestJSON.body = await BatchRequestContent.getBody(request);
+        }
+        return requestJSON;
+    }
+
+    /**
+     * @static
+     * @async
+     * Gets the body of a Request object instance
+     * @param {Request} request - The Request object instance
+     * @return The Promise that resolves to a body value of a Request
+     */
+    static async getBody(request: Request): Promise<any> {
+        let bodyParsed: boolean = false,
+            body;
+        try {
+            let cloneReq = request.clone();
+            body = await cloneReq.json();
+            bodyParsed = true;
+        } catch (e) {
+            
+        }
+        if (!bodyParsed) {
+            try {
+                let blob = await request.blob(),
+                    reader = new FileReader();
+                body = await new Promise(resolve => {
+                    reader.addEventListener("load", function () {
+                        resolve(reader.result);
+                    }, false);
+                    reader.readAsDataURL(blob);
+                });
+                bodyParsed = true;
+            } catch (e) {
+                
+            }
+        }
+        return body;
+    }
+
+    /**
+     * @static
+     * To get the random id
+     * @return The random id of length 9
+     */
+    static getRandomId(): string {
+        return Math.random().toString(36).substr(2, 9);
+    }
+
+    /**
+     * @async
+     * Adds a request to the batch payload
+     * @param {any} request - Request object or a json representing request
+     * @return The id of the added request (id will be generated in case if user didn't provide one)
+     */
+    async addRequest(request: any): Promise<any> {
+        let self = this,
+            limit = BatchRequestContent.requestLimit;
+        if (self.requests.size === limit) {
+            let error = new Error(`Maximum requests limit exceeded, Max allowed numbers of requests are ${limit}`);
+            error.name = "Limit Exceeded Error";
+            throw error;
+        }
+        if (request.constructor.name === "Request") {
+            request = await BatchRequestContent.getJSONFromRequest(<Request>request);
+        }
+        if (request.id === undefined) {
+            request.id = BatchRequestContent.getRandomId();
+        }
+        this.requests.set(request.id, request);
```

Comment:
First, `BatchRequestContent.getRandomId();` will surely give us the good randomness, I am pretty confident that we won't get collision. So we don't have to worry about overwrite in case of system generated id.
Second, lets say user is providing the key for the requests, in this case I thought of overwriting the requests user might want to. instead either overwriting or creating new random id, we could throw error.

Reply:
Yeah, we should throw an error if they are providing a key that collides with an existing key. They could be using the provided value so providing an unexpected random ID would not be helpful in that scenario.